src/partition/repart.c

```diff
@@ -7115,12 +7132,23 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
-                case ARG_PRIVATE_KEY_URI: {
-                        r = free_and_strdup_warn(&private_key_uri, optarg);
+                case ARG_PRIVATE_KEY_SOURCE:
+                        if (startswith(optarg, "file:"))
+                                arg_private_key_source_type = KEY_SOURCE_FILE;
+                        else if (startswith(optarg, "engine:"))
+                                arg_private_key_source_type = KEY_SOURCE_ENGINE;
+                        else if (startswith(optarg, "provider:"))
+                                arg_private_key_source_type = KEY_SOURCE_PROVIDER;
+                        else
+                                return log_error_errno(
+                                                SYNTHETIC_ERRNO(EINVAL),
+                                                "Invalid private key source '%s'",
+                                                optarg);
+
+                        r = free_and_strdup_warn(&arg_private_key_source, optarg);
```

Comment:
this variable should only contain the suffix, and only in the engine/provider case.

ie.:

```c
const char *e = NULL;

if (streq(optarg, "file")) 
        arg_private_key_source_type = KEY_SOURCE_FILE;
else if (startswith(optarg, "engine:")) 
        arg_private_key_source_type = KEY_SOURCE_ENGINE;
else if (startswith(optarg, "provider:"))
        arg_private_key_source_type = KEY_SOURCE_PROVIDER;
else 
        return …;

r = free_and_strdup_warn(&arg_private_key_source, e);
…
```

Reply:
maybe even turn this into a new helper in openssl-util.h that is modelled after those from src/shared/parse-argument.h

after all you need this exactly the same way in systemd-measure